mesa/time.py

```diff
@@ -102,25 +112,31 @@ def get_agent_count(self) -> int:
         return len(self._agents)
 
     @property
-    def agents(self) -> list[Agent]:
-        return list(self._agents.values())
+    def agents(self) -> AgentSet:
+        # a bit dirty, but returns a copy of the internal agent set
+        return self._agents.select()
 
     def get_agent_keys(self, shuffle: bool = False) -> list[int]:
         # To be able to remove and/or add agents during stepping
         # it's necessary to cast the keys view to a list.
-        agent_keys = list(self._agents.keys())
+
+        if not self._agents_key_warning_given:
+            self._agents_key_warning_given = True
+            warnings.warn(
+                "Because of the shift to using weakrefs, this method will be removed in a future version",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+
+        agent_keys = [agent.unique_id for agent in self._agents]
```

Comment:
Agreed on deprecating in this case.

However, while in this way it's probably fine, we're technically deprecating something for which the replacement (AgentSet) is still experimental. Might be something to keep in mind.

Reply:
Again, this is outside the scope of this PR, but I want to do a separate discussion on unique_id. Now that it is no longer needed for the schedulers, its only function is to identify agents when gathering statistics.

Reply:
Agree - outside of this pr - suggested place for the discussion?

Reply:
Given that #1930 also relies on unique_id, we can perhaps discuss it there. Alternatively, we can have a broader discussion on the future of data collection and make it part of that.